src/libraries/System.Private.CoreLib/src/System/Numerics/Matrix4x4.cs

```diff
@@ -1311,8 +1311,180 @@ public static Matrix4x4 CreateReflection(Plane value)
         /// <param name="matrix">The source matrix to invert.</param>
         /// <param name="result">If successful, contains the inverted matrix.</param>
         /// <returns>True if the source matrix could be inverted; False otherwise.</returns>
-        public static bool Invert(Matrix4x4 matrix, out Matrix4x4 result)
+        public static unsafe bool Invert(Matrix4x4 matrix, out Matrix4x4 result)
         {
+            if (Sse.IsSupported)
```

Comment:
Was this ported from the DirectX Math implementation?

If so, we need to document that and add some third party notices.

Reply:
I can add a comment and update the third party notices page.   One other thing I wanted to add is the test case for when the determinant is zero, currently that path is not being tested.